src/code/components/modal-alert.js

```diff
@@ -77,6 +80,7 @@ class ModalAlert extends React.Component {
               onHide={this.props.onHide} >
         <div style={dialogStyle()} >
           <h4 id='modal-label'>{this.props.message}</h4>
+          <AwardView challengeProgress={this.props.challengeProgress} />
```

Comment:
This works, but is a little confusing to me because it looks like an AwardView is always expected to be rendered, but then AwardView itself might return false.

Another way could be 

    if (this.props.challengeProgress) {
        <AwardView challengeProgress={this.props.challengeProgress} />
    }

I don't know if one is a more idiomatic React convention than the other, so I'm also fine leaving it as is.